src/vue/modules/buy-back-form/index.vue

```diff
@@ -0,0 +1,289 @@
+          </div>
+        </template>
+      </form>
+    </div>
+    <no-data-message
+      :title="'buy-back-form.no-investments' | globalize"
+      :message="'buy-back-form.here-will-investments-list' | globalize"
+      v-else-if="!assetsInBalance.length && isInitialized"
+    />
+    <loader v-else message-id="buy-back-form.loading-msg" />
+  </div>
+</template>
+
+<script>
+import { mapActions, mapMutations, mapGetters } from 'vuex'
+import { types } from './store/types'
+
+import { Wallet } from '@tokend/js-sdk'
+import { initApi } from './_api'
+
+import FormConfirmation from '@/vue/common/FormConfirmation'
+import Loader from '@/vue/common/Loader'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+
+import FormMixin from '@/vue/mixins/form.mixin'
+import OfferManagerMixin from '@/vue/mixins/offer-manager.mixin'
+
+import { MathUtil } from '@/js/utils/math.util'
+
+import {
+  required,
+  maxValue,
+  minValue,
+  maxDecimalPoints,
+} from '@validators'
+
+const EVENTS = {
+  transferred: 'transferred',
+}
+
+export default {
+  name: 'buy-back-form',
+  components: {
+    FormConfirmation,
+    NoDataMessage,
+    Loader,
+  },
+  mixins: [FormMixin, OfferManagerMixin],
+  props: {
+    wallet: {
+      type: Wallet,
+      required: true,
+    },
+    /**
+     * @property config - the config for component to use
+     * @property config.horizonURL - the url of horizon server (without version)
+     * @property config.minAmount - min allowed amount
+     * @property config.decimalPoints - default max allowed points after dot
+     * @property [config.defaultAssetCode] - prefills the asset-selector with
+     *           this asset code
+     */
+    config: {
+      type: Object,
+      required: true,
+    },
+  },
+  validations () {
+    return {
+      form: {
+        amount: {
+          required,
+          maxDecimalPoints: maxDecimalPoints(this.config.decimalPoints),
+          maxValue: maxValue(this.allowedToBuy(this.form.asset.code)),
+          minValue: minValue(0),
+        },
+        totalAmount: {
+          minValue: minValue(this.config.minAmount),
+        },
+      },
+    }
+  },
+  data: _ => ({
+    isInitialized: false,
+    form: {
+      asset: {},
+      quoteAsset: {},
+      amount: '',
+      totalAmount: '',
+    },
+    selectedSale: {},
+    isBuyBackProcessing: false,
+  }),
+  computed: {
+    ...mapGetters('buy-back-form', {
+      balances: types.balances,
+      assets: types.assets,
+      assetsInBalance: types.assetsInBalance,
+      allowedToBuy: types.allowedToBuy,
+    }),
+  },
+  watch: {
+    async 'form.asset' (asset) {
+      this.disableForm()
+      this.selectedSale = await this.loadSaleByBaseAsset(asset.code)
+      this.form.quoteAsset = this.selectedSale.quoteAssets[0]
+      this.enableForm()
+    },
+    selectedSale (sale) {
+      this.form.amount = this.allowedToBuy(this.form.asset.code)
+      this.form.totalAmount = this.calculatedBuyBackPrice()
```

Comment:
is is price or amount? Is not it strange that `calculatedBuyBackPrice` returns a `totalAmount` value?